mitodl/ocw-studio
# separate concourse web and worker into two containers

# What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1924

# Description (What does it do?)
This PR moves the `concourse` node in our `docker-compose.yml` file to use the `web` command instead of `quickstart`, and adds a new node using the same image that uses the `worker` command, spawning the web app and worker on separate containers. This is being done so that settings that apply only to the worker, like `CONCOURSE_CONTAINERD_MAX_CONTAINERS` can be set. The version of Concourse was also updated to reflect the latest version.

# How can this be tested?
 - First of all, change the `CONCOURSE_CONTAINERD_MAX_CONTAINERS` setting in `docker-compose.yml` temporarily to 1
 - Spin up all containers with `docker-compose up`
 - If you do not have any pipeline definitions in your local Concourse instance, push up the theme assets pipeline with `docker-compose exec web ./manage.py upsert_theme_assets_pipeline`
 - If you unpause the pipeline and click into it, you should see an error pretty much immediately, but if you don't trigger a build
 - Verify that you see an error, either on the `ocw-hugo-themes` input or on the build after triggering it that looks something like `run check: find or create container on worker 9ea1c54b4a56: new container: checking container capacity: max containers reached`
 - Change the `CONCOURSE_CONTAINERD_MAX_CONTAINERS` setting in `docker-compose.yml` back to 0
 - Restart your containers and re-run the above process, the error should go away

# Additional Context
This problem was actually discovered while working on https://github.com/mitodl/ocw-studio/issues/1924. The default limit of 250 containers would be hit while running the new mass build pipeline being worked on there, and the `CONCOURSE_CONTAINERD_MAX_CONTAINERS` setting was doing nothing. This fixes that.
